c10/core/ModePyObjTrampoline.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <c10/core/SafePyObject.h>
+
+namespace c10 {
+
+// this is basically just a light wrapper around a SafePyObject that also has
```

Comment:
Is there a reason we don't use PyInterpreter for this?

Reply:
Not sure I fully get the question/suggestion, but here's the context. The basis of this was that we needed to call a specific python function (check_mode_state_push/pop) from TorchDispatchModeTLS (which is in c10). The assumption was that c10 didn't want to have a dependency on pybind or any of these things in c10.

So this builds a trampoline that ends up sending a SafePyObject back to the torch namespace. There torch grabs the current PyInterpreter and calls the specific Python function

So the trampoline lets c10 call a function that it knows should be defined on the mode object without making it depend on pybind

Reply:
Yes. What I am saying is, PyInterpreter is precisely this trampoline mechanism. It has a vtable with a big grab bag of things you might want to do to PyObject. You can just add a pair of new virtual methods to it.

Reply:
Ah I see--I didn't love the look of getting the interpreter from an object just to have to pass in the object (e.x.     
 `mode->interpreter()->mode_state_push_trampoline(mode)`) but since that's where all the trampolines are, moved it over